ci/docker/conda-integration.dockerfile

```diff
@@ -72,3 +70,102 @@ ENV ARROW_BUILD_INTEGRATION=ON \
     ARROW_S3=OFF \
     ARROW_USE_GLOG=OFF \
     CMAKE_UNITY_BUILD=ON
+
+RUN /arrow/ci/scripts/cpp_build.sh /arrow /build
+
+##################### C# #####################
+FROM ${repo}:${arch}-conda-cpp AS builder-csharp
+
+ENV DOTNET_ROOT=/opt/dotnet \
+    PATH=/opt/dotnet:$PATH
+RUN curl -sSL https://dot.net/v1/dotnet-install.sh | bash /dev/stdin -Channel 3.1 -InstallDir /opt/dotnet
+
+COPY ./csharp /arrow/csharp
+COPY ./format /arrow/format
+COPY ./ci/scripts/csharp_build.sh /arrow/ci/scripts/csharp_build.sh
+
+RUN /arrow/ci/scripts/csharp_build.sh /arrow /build
+
+##################### JS #####################
+FROM ${repo}:${arch}-conda-cpp AS builder-js
+
+RUN mamba install -q \
+        yarn && \
+    mamba clean --all --force-pkgs-dirs
+
+COPY ./js /arrow/js
+COPY ./LICENSE.txt /arrow/js/
+COPY ./NOTICE.txt /arrow/js/
+COPY ./ci/scripts/js_build.sh /arrow/ci/scripts/js_build.sh
+
+RUN /arrow/ci/scripts/js_build.sh /arrow /build
+
+##################### Java #####################
+FROM ${repo}:${arch}-conda-cpp AS builder-java
+
+ARG maven=3.5
+ARG jdk=8
+RUN mamba install -q \
+        maven=${maven} \
+        openjdk=${jdk} && \
+    mamba clean --all --force-pkgs-dirs
+
+COPY ./java /arrow/java
+COPY ./format /arrow/format
+COPY ./ci/scripts/java_build.sh /arrow/ci/scripts/java_build.sh
+
+RUN /arrow/ci/scripts/java_build.sh /arrow /build
+
+##################### RUST ######################
+FROM ${repo}:${arch}-conda-cpp AS builder-rust
```

Comment:
The only concern I have is that the today we don't require to have Rust cloned locally but I am not sure how we can skip a stage on docker or how we could make this conditional following this approach
As expected if I try to build locally I get the error for Rust:
```
------                                                                                                                                                         
 > [builder-rust 3/5] COPY ./rust /arrow/rust:                                                                                                                 
------                                                                                                                                                         
failed to solve: failed to compute cache key: "/rust" not found: not found 
```